test/unit/auth/auth-api-request.spec.ts

```diff
@@ -3549,6 +3550,552 @@ AUTH_REQUEST_HANDLER_TESTS.forEach((handler) => {
+            AuthClientErrorCode.INTERNAL_ERROR,
+            'INTERNAL ASSERT FAILED: Unable to create new tenant',
+          );
+          const stub = sinon.stub(HttpClient.prototype, 'send').resolves(utils.responseFrom({}));
+          stubs.push(stub);
+
+          const requestHandler = handler.init(mockApp) as AuthRequestHandler;
+          return requestHandler.createTenant(tenantOptions)
+            .then((resp) => {
+              throw new Error('Unexpected success');
+            }, (error) => {
+              expect(error).to.deep.equal(expectedError);
+              expect(stub).to.have.been.calledOnce.and.calledWith(callParams(path, method, expectedRequest));
+            });
+        });
+
+        it('should be rejected when the backend returns a response missing tenant ID in response name', () => {
+          const expectedError = new FirebaseAuthError(
+            AuthClientErrorCode.INTERNAL_ERROR,
+            'INTERNAL ASSERT FAILED: Unable to create new tenant',
+          );
+          // Resource name should have /tenants/tenant_id in path. This should throw an error.
+          const stub = sinon.stub(HttpClient.prototype, 'send')
+            .resolves(utils.responseFrom({name: 'projects/project_id'}));
+          stubs.push(stub);
+
+          const requestHandler = handler.init(mockApp) as AuthRequestHandler;
+          return requestHandler.createTenant(tenantOptions)
+            .then((resp) => {
+              throw new Error('Unexpected success');
+            }, (error) => {
+              expect(error).to.deep.equal(expectedError);
+              expect(stub).to.have.been.calledOnce.and.calledWith(callParams(path, method, expectedRequest));
+            });
+        });
+
+        it('should be rejected when the backend returns an error', () => {
+          const expectedServerError = utils.errorFrom({
+            error: {
+              message: 'INTERNAL_ERROR',
+            },
+          });
+          const expectedError = new FirebaseAuthError(
+            AuthClientErrorCode.INTERNAL_ERROR,
+            `An internal error has occurred. Raw server response: ` +
+            `"${JSON.stringify(expectedServerError.response.data)}"`,
+          );
+          const stub = sinon.stub(HttpClient.prototype, 'send').rejects(expectedServerError);
+          stubs.push(stub);
+
+          const requestHandler = handler.init(mockApp) as AuthRequestHandler;
+          return requestHandler.createTenant(tenantOptions)
+            .then((resp) => {
+              throw new Error('Unexpected success');
+            }, (error) => {
+              expect(error).to.deep.equal(expectedError);
+              expect(stub).to.have.been.calledOnce.and.calledWith(callParams(path, method, expectedRequest));
+            });
+        });
+      });
+
+      describe('updateTenant', () => {
+        const path = '/v2beta1/projects/project_id/tenants/tenant_id';
+        const method = 'PATCH';
```

Comment:
May be call this `patchMethod`? That will make the code that uses it more readable:

```
expect(stub).to.have.been.calledOnce.and.calledWith(callParams(expectedPath, patchMethod, expectedRequest));
```

Reply:
done